### ISSUE TITLE
Class body hit character limit 10000, Apex Language Server stops working, breaking replay debugger, etc.

### ISSUE THREAD

**Issue by zerkz** [forcedotcom/salesforcedx-vscode#5232]
### Summary
Get "Class body hit character limit 10000" error in Apex Language Server output, and then a variety of other explosions/exceptions after. These additional exceptions do not make any sense, stuff like "Class.init" is already defined, but its a constructor of a class which is perfectly valid. It seems the limit 10000 bug causes the server to malfunction afterwards.

Class was roughly 22k characters (including spaces?). Some nested classes within this one (included them in the character count). Class is deployed and compiled in production already. I cannot provide the exact class source.

With this error occurring, it is impossible to start the Apex Replay Debugger for completely unrelated classes/apex. 

```
09:54:28.487 [ApexIndexer STARTING] INFO  a.jorje.lsp.impl.index.ApexIndexer - ApexIndexer: Used Memory (MB): 79
09:54:28.487 [ApexIndexer STARTING] INFO  a.jorje.lsp.impl.index.ApexIndexer - ApexIndexer: Scanning user-defined types took 1316 ms.
Nov 15, 2023 9:54:53 AM apex.jorje.semantic.compiler.CodeUnit traverse
INFO: unexpected error during apex MemberResolveVisitor:
Class body hit character limit 10000 public with sharing class SingleViewController {
```

### Steps To Reproduce:
N/A

### Expected result
There is no documented character limit around Apex class body. Is there a documentation miss? 

### Actual result
I'd expect Apex class body size to not cause the Apex Language Server to malfunction and prevent other features to work.

### Additional information

**Salesforce Extension Version in VS Code**:

Salesforce Extension Pack - v59.4.0
Salesforce Apex plugin - v59.4.0
Also tried with the latest jorje JAR from the develop branch, still broken.

**SFDX CLI Version**:

@salesforce/cli/2.17.14 darwin-arm64 node-v18.18.1

**OS and version**:

macOS Ventura 13.6

**VS Code version**:

Version: 1.83.1 (Universal)

**Comment by peternhale** [comment]
@zerkz Yes, the limit is 1 million characters. The exception you are seeing is during a AST traversal of the Apex class in question. The 10k limit is on how much source to include in the exception, which is imposed by the exception being thrown.

Is there a version of our extension where you were able to start debugger w/o this error?
Is the large cls file open an editor tab as you start debugger?
How large is your project, approximate number of Apex related files, .cls, .trigger, .apex?

Could you try:

1. close all open editor tabs and shutdown vscode
2. `rm -fr .sfdx/tools/246` - this will force a full index of your workspace
3. open your project with vscode
4. open on .cls file (not the large one, so you can see the indexing status bar item)
5. wait for indexing complete
6. start apex debugger

I would also help to have you share the entire output tab for "Apex Language Server"

**Comment by zerkz** [comment]
Ah, that makes more sense. I will seek out the true exception. 

I did not have the file open in my editor when I tried to start the debugger. 

Deleting the class file (`SingleViewController`) in my repository (locally) solved the issue/worked around it. 

After debugging once with a log file, I undeleted the class and did `Reload Windows` via VSCode, the apex language server is not throwing any errors now. 

I am guessing the index was regenerated or fixed. 

Thanks for the info @peternhale  -- I will close this issue since the 10k limit is just how much source to show in the error (and not the actual error it is encountering).
